### PR TITLE
Websockets transport now creates correct url

### PR DESCRIPTION
**Pull request by mikaelm12**
Addresses: https://github.com/aspnet/SignalR/issues/266

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public EndToEndTests(ServerFixture serverFixture)
 
         [ConditionalFact]
         [OSSkipCondition(OperatingSystems.Windows, WindowsVersions.Win7, WindowsVersions.Win2008R2, SkipReason = "No WebSockets Client for this platform")]
-        public async Task WebSocketsTest()
+        public async Task BareWebSocketsTest()
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: Not sure why the test name was changed...

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 2]
```diff
@@ -100,21 +110,22 @@ public static IEnumerable<object[]> MessageSizesData
         {
             get
             {
-                yield return new object[] { new string('A', 5 * 1024)};
-                yield return new object[] { new string('A', 5 * 1024 * 1024 + 32)};
+                yield return new object[] { new string('A', 5 * 1024), new WebSocketsTransport()};
+                yield return new object[] { new string('A', 5 * 1024 * 1024 + 32), new WebSocketsTransport() };
+                yield return new object[] { new string('A', 5 * 1024), new LongPollingTransport(new HttpClient())};
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: This does not seem right (given the name of the property). If needed at all (and I am not convinced it is) you should create a getter that creates a Cartesian product of what before was `MessageSizesData` and `Transports`

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 3]
```diff
@@ -100,21 +110,22 @@ public static IEnumerable<object[]> MessageSizesData
...
+                yield return new object[] { new string('A', 5 * 1024 * 1024 + 32), new LongPollingTransport(new HttpClient())};
             }
         }
 
         [ConditionalTheory]
         [OSSkipCondition(OperatingSystems.Windows, WindowsVersions.Win7, WindowsVersions.Win2008R2, SkipReason = "No WebSockets Client for this platform")]
         [MemberData(nameof(MessageSizesData))]
-        public async Task ConnectionCanSendAndReceiveDifferentMessageSizesWebSocketsTransport(string message)
+        public async Task ConnectionCanSendAndReceiveDifferentMessageSizesWebSocketsTransport(string message, ITransport transport)
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: This test (as indicated by its name) is used to test the logic we have in the WebSockets transport to combine messages which don't fit the buffer/have endOfMessage flag set to false. As such this test is specific to WebSocketsTransport and doesn't have to be run against other transports. We already have a test above that test that sending and receiving data using different transports works.

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs (hunk 1 above)
Comment (mikaelm12) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: Just to be a little more explicit that this test is using bare websockets and not testing the transports. I can change this back

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: Yeah, change it back. Other than that it now looks good.

**Comment by mikaelm12** [comment]
Hmm. Travis Failure https://travis-ci.org/aspnet/SignalR/jobs/208377625#L2744
Any help on this @anurse

**Comment by anurse** [comment]
\*sigh\* it's something timing out on Travis yet again. I can't figure out why these tests are so flaky only on Travis. It's got to be something about slow machines and task scheduling there, but it's almost impossible to reproduce on any of my normal machines. For now, restarting the build (which I just did) normally works.

**Pull request merged**